Source/Scene/Cesium3DTileset.js

```diff
@@ -739,10 +742,19 @@ define([
                 that._extensionsUsed = tilesetJson.extensionsUsed;
                 that._gltfUpAxis = gltfUpAxis;
                 that._extras = tilesetJson.extras;
-                if (!defined(tilesetJson.root.transform)) {
-                    that._useBoundingSphereForClipping = true;
-                    that._clippingPlaneOffsetMatrix = Transforms.eastNorthUpToFixedFrame(that.boundingSphere.center);
+                // Save the original, untransformed bounding volume position so we can apply
+                // the tile transform and model matrix at run time
+                var boundingVolume = that._root.createBoundingVolume(tilesetJson.root.boundingVolume, Matrix4.IDENTITY);
+                var clippingPlanesOrigin = boundingVolume.boundingSphere.center;
+                // If this origin is above the surface of the earth and root transform isn't identity
+                // we want to apply an ENU orientation as our best guess of orientation.
+                // Otherwise, we assume it gets its position/orientation completely from the
+                // root tile transform and the tileset's model matrix
+                var heightAboveEllipsoid = Cartographic.fromCartesian(clippingPlanesOrigin).height;
```

Comment:
I can see that other parts of the engine do this, but doesn't it make sense to use the actual ellipsoid that's being used instead of the default one? For example, if you were using the moon, or something much smaller, using that ellipsoid here would work perfectly whereas it would no longer be a correct measure of above/below the surface with the default ellipsoid.

Reply:
Also, it looks like it already uses WGS84 in this case if ellipsoid isn't passed as a parameter in `fromCartesian`, so that should be the correct behavior right?

Reply:
**Always** pass in the ellipsoid.  We have a bunch of bugs for non-earth ellipsoids because (especially) at the entity layer we aren't passing the right ellipsoid around.  I recommend not using these `Cartographic.fromCartesian`, `Cartographic.toCartesian`, `Cartesian3.fromDegrees` helper functions in code we develop because while they're convenient for our end users, it's easy to forget to pass in the ellipsoid.  I always use `ellipsoid.cartesianToCartographic` and `ellipsoid.cartographicToCartesian`.

Reply:
Oh, I think I misunderstood. So we _do_ want to pass in the ellipsoid the user is using. Thanks for clarifying that!